RecoverCRPVM/CreateCRPRescueVM.ps1

```diff
@@ -0,0 +1,260 @@
+        [String]$ResourceGroup,
+
+        [Parameter(mandatory=$true)]
+        [String]$SubID,
+
+        [Parameter(mandatory=$false)]
+        [Bool]$showErrors=$true,
+
+        [Parameter(mandatory=$false)]
+        [String]$prefix = "rescue",
+
+        [Parameter(mandatory=$false)]
+        [String]$Sku,
+
+        [Parameter(mandatory=$false)]
+        [String]$Offer,
+
+        [Parameter(mandatory=$false)]
+        [String]$Publisher,
+
+        [Parameter(mandatory=$false)]
+        [String]$Version
+     )
+
+#windows
+#Sample Execution (Unmanaged)          ==> .\CreateCRPRescueVM.ps1 -ResourceGroup sujtemp -VmName sujnortheurope -SubID d7eaa135-abdf-4aaf-8868-2002dfeea60c
+#Sample execution (managed) Windows VM ==> .\CreateCRPRescueVM.ps1 -VmName SujasManagedVM -ResourceGroup rescueMgSujasRG -SubID d7eaa135-abdf-4aaf-8868-2002dfeea60c --Current Version NOT SUPPORTED
+#                                      ==> .\CreateCRPRescueVM.ps1 -ResourceGroup rescueMgScriptSujas -VmName SujasWinManagedVM -SubID d7eaa135-abdf-4aaf-8868-2002dfeea60c
+
+#Linux
+#Linux VM                             ==> .\CreateCRPRescueVM.ps1 -VmName myLinuxVM -ResourceGroup sujwithavsetwe -SubID d7eaa135-abdf-4aaf-8868-2002dfeea60c  --Current Version NOT SUPPORTED
+#                                     ==> .\CreateCRPRescueVM.ps1 -ResourceGroup rescueportalLin -VmName ubuntu -SubID d7eaa135-abdf-4aaf-8868-2002dfeea60c
+#/subscriptions/d7eaa135-abdf-4aaf-8868-2002dfeea60c/resourceGroups/rescueportalLin/providers/Microsoft.Compute/virtualMachines/ubuntu
+$Error.Clear()
+cls
+if (-not $showErrors) {
+    $ErrorActionPreference = 'SilentlyContinue'
+}
+
+$script:scriptStartTime = (Get-Date).ToUniversalTime()
+$LogFile = $env:TEMP + "\" + $VmName + "_" + ( Get-Date $script:scriptStartTime -f yyyyMMddHHmmss ) + ".log"  
+# Get running path
+$RunPath = split-path -parent $MyInvocation.MyCommand.Source
+cd $RunPath
+$CommonFunctions = $runPath+"\Common-Functions.psm1"
+
+#Import-Module Common-Functions -ArgumentList $LogFile -ErrorAction Stop 
+if (Get-Module Common-Functions) {remove-module -name Common-Functions}   
+Import-Module -Name $CommonFunctions  -ArgumentList $LogFile -ErrorAction Stop 
+write-log "Info: Log is being written to ==> $LogFile" 
+
+
+
+if (-not (Get-AzureRmContext).Account)
+{
+    Login-AzureRmAccount
+}
+
+#Set the context to the correct subid
+Write-Log "Setting the context to SubID $SubID" -color Yellow
+$subContext= Set-AzureRmContext -Subscription $SubID
+write-log $subContext -logOnly
+if ($subContext -eq $null) 
+{
+    Write-Log "Unable to set the Context for the given subId ==> $SubID, Please make sure you first  run the command ==> Login-AzureRMAccount" -Color Red
+    return
+}
+
+
+#Step 1 Get the VM Object
+Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$ResourceGroup`" -Name `"$VmName`"" -Color Yellow
+$vm = Get-AzureRmVM -ResourceGroupName $ResourceGroup -Name $VmName 
+write-log "`"$vm`" => $($vm)" -logOnly
```

Comment:
Doubt about PS: In many places we do write-log and then check the variable. So is null handled (because cmd failed) ?